### PR TITLE
attach picture button in forum post dialog; message size check

### PR DESCRIPTION
**Pull request by RetroPooh**

### CODE REVIEW

File: retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp [hunk 1]
```diff
@@ -337,6 +340,20 @@ void  CreateGxsForumMsg::loadFormInformation()
 	//ui.forumMessage->setText("");
 }
 
+void CreateGxsForumMsg::checkLength()
+{
+	QString text;
+	RsHtml::optimizeHtml(ui.forumMessage, text);
+	std::wstring msg = text.toStdWString();
+	int charRemains = 199000 - msg.length(); // MAX_ALLOWED_GXS_MESSAGE_SIZE
```
Comment (hunbernd) [inline retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp]: Do not hardcode the value, use the MAX_ALLOWED_GXS_MESSAGE_SIZE instead!

File: retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp [hunk 2]
```diff
@@ -487,6 +504,18 @@ void CreateGxsForumMsg::addFile()
 	}
 }
 
+void CreateGxsForumMsg::addPicture()
+{
+	QString file;
+	if (misc::getOpenFileName(window(), RshareSettings::LASTDIR_IMAGES, tr("Load Picture File"), "Pictures (*.png *.xpm *.jpg *.jpeg)", file)) {
+		QString encodedImage;
+		if (RsHtml::makeEmbeddedImage(file, encodedImage, 640*480, 199000 - 200)) {		// MAX_ALLOWED_GXS_MESSAGE_SIZE -200 for the html stuff
```
Comment (hunbernd) [inline retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp]: Do not hardcode the value, use the MAX_ALLOWED_GXS_MESSAGE_SIZE instead!

**Review by hunbernd — CHANGES_REQUESTED** [review]

File: retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp (hunk 1 above)
Comment (csoler) [reply retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp]: yes, please update as hunbernd suggests.

File: retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp [hunk 3]
```diff
@@ -337,6 +340,22 @@ void  CreateGxsForumMsg::loadFormInformation()
 	//ui.forumMessage->setText("");
 }
 
+static const uint32_t MAX_ALLOWED_GXS_MESSAGE_SIZE = 199000;
```
Comment (PhenomRetroShare) [inline retroshare-gui/src/gui/gxsforums/CreateGxsForumMsg.cpp:343]: Already defined constant.
Prefer adding method to get it like was done for chat:
ChatWidget::maxMessageSize()
p3Msgs::getMaxMessageSecuritySize(int type)
p3ChatService::getMaxMessageSecuritySize(int type)

**Comment by defnax** [comment]
this is a nice idea, hope it can be soon merged when its ready?
i had some time ago same idea add a button for attach pictures for forums.

**Comment by defnax** [comment]
how long wait to be merged? it always needed this feature

**Comment by defnax** [comment]
whats the status ? @RetroPooh @hunbernd @PhenomRetroShare

**Comment by defnax** [comment]
> _No description provided._

any news about this pr? soon 3 month isw gone
@RetroPooh

**Comment by defnax** [comment]
any news? when it will be merged?

**Comment by defnax** [comment]
whats the status?

**Comment by defnax** [comment]
this is a needed feature im not sure why it will be not merged

**Comment by defnax** [comment]
ok i checked seems code isnt more up to date

**Comment by felisucoibi** [comment]
+1

**Pull request closed without merging**